.github/workflows/integration_test.yaml

```diff
@@ -21,24 +22,88 @@ jobs:
+          echo -e "# Charm Upload Automation Test Index Title some uuid: $(uuidgen)\n" \
+            "placeholder index content for testing, " \
             "repository: ${{ github.repository }}, " \
             "branch: ${{ github.head_ref }}, " \
             "commit sha: ${{ github.sha }}, " \
             "some uuid: $(uuidgen)" \
             > docs/index.md
-      - name: Self test
-        id: selfTest
+          echo -e "# Charm Upload Automation Test Documentation Title some uuid: $(uuidgen)\n" \
+            "placeholder documentation content for testing, " \
+            "repository: ${{ github.repository }}, " \
+            "branch: ${{ github.head_ref }}, " \
+            "commit sha: ${{ github.sha }}, " \
+            "some uuid: $(uuidgen)" \
+            > docs/doc.md
+          echo -e "# Charm Upload Automation Test Documentation Alternate Title some uuid: $(uuidgen)\n" \
+            "placeholder documentation alternate content for testing, " \
+            "repository: ${{ github.repository }}, " \
+            "branch: ${{ github.head_ref }}, " \
+            "commit sha: ${{ github.sha }}, " \
+            "some uuid: $(uuidgen)" \
+            > docs/alternate_doc.md
+      - name: Draft self test
+        id: selfTestDraft
         uses: ./
         with:
-          create_new_topic: true
           discourse_host: discourse.charmhub.io
           discourse_api_username: ${{ secrets.DISCOURSE_API_USERNAME }}
           discourse_api_key: ${{ secrets.DISCOURSE_API_KEY }}
-      - name: Show created pages
-        run: echo '${{ steps.selfTest.outputs.created_pages }}'
-      - name: Clean up
+          draft_mode: true
+      - name: Show pages
+        run: echo '${{ steps.selfTestDraft.outputs.urls_with_actions }}'
+      - name: Check draft
         run: |
           sudo apt update && sudo apt install python3-pip
           pip3 install -r requirements.txt
-          ./discourse_cleanup.py '${{ steps.selfTest.outputs.created_pages }}' '${{ steps.selfTest.outputs.discourse_config }}'
+          ./discourse_check_cleanup.py --check-draft --check-only '${{ steps.selfTestDraft.outputs.urls_with_actions }}' '${{ steps.selfTestDraft.outputs.discourse_config }}'
+      - name: Create self test
+        id: selfTestCreate
+        uses: ./
+        with:
+          discourse_host: discourse.charmhub.io
+          discourse_api_username: ${{ secrets.DISCOURSE_API_USERNAME }}
+          discourse_api_key: ${{ secrets.DISCOURSE_API_KEY }}
+      - name: Show pages
+        run: echo '${{ steps.selfTestCreate.outputs.urls_with_actions }}'
+      - name: Check create
+        run: |
+          ./discourse_check_cleanup.py --check-create --check-only '${{ steps.selfTestCreate.outputs.urls_with_actions }}' '${{ steps.selfTestCreate.outputs.discourse_config }}'
+      - name: Add docs key to metadata
+        run: |
+          echo "docs: ${{ steps.selfTestCreate.outputs.index_url }}" >> metadata.yaml
```

Comment:
I have the feeling that this is a little obscure and TBH I don't like much the idea of modifying the code automatically. I guess we can't post to the same URL and upgrade the content each version, right?

Reply:
This is an integration test for the action only, it has no effect on when the action is run by a user. There are plans for future features where the action creates a new PR to add the docs key back if needed: https://github.com/canonical/upload-charm-docs/issues/8